### DISCUSSION TITLE
I have some questions regarding this game engine

### DISCUSSION THREAD

**Discussion by Joe23232** [o3de/o3de#6368] [General]
Hey guys I have some questions regarding this game engine:

1. Is it completely free and open source, I can even disclose my game product using commercial closed source license?
2. Can I code in C?
3. Is it a modular game engine, like ecs?
4. Can I make 2D games as well?

**Comment by soltsmank** [comment]
Hey Joe,

The engine (O3DE) was sourced out from Lumberyard,  which was AWS owned, but was FREE for use to create commercial projects. You don't pay money for the engine usege.
O3DE is under Apache 2.0 license. You can do almost whatever you want, but you can't rebrand and sell the engine.

The engine handles 3D space, you can make everything in 2 axises as a workaround. I didn't seen any 2D special components. Not sure even contributors will go that way to add it.

Regards.

**Reply by Raj2032** [reply]
Hey mate, thanks for your response,

When I use C++, does it get statically compiled with the game engine code?

**Comment by DavidHammertim** [comment]
The [ScriptCanvas Workshop](https://youtu.be/GYXG4nrQPnE?t=1295) is good if you want an idea on how a 2D game can be made in O3DE.

**Reply by Raj2032** [reply]
THanks mate
